hoomd/mpcd/DryingDropletStreamingMethod.h

```diff
@@ -0,0 +1,271 @@
+
+        //! For Making a random pick of particles across all ranks
+        void makeAllPicks(unsigned int timestep, unsigned int N_pick_total, unsigned int N_bounced_total);
+    };
+
+/*!
+ * \param timestep Current time to stream
+ */
+
+void DryingDropletStreamingMethod::stream(unsigned int timestep)
+    {
+    //compute final Radius and Velocity of surface
+    const Scalar start_R = m_R->getValue(timestep);
+    const Scalar end_R = m_R->getValue(timestep + m_period);
+    const Scalar V = (end_R - start_R)/(m_period * m_mpcd_dt);
+    
+    if (V > 0)
+        {
+        throw std::runtime_error("Droplet radius must decrease.");
+        }
+    
+    /*
+     * If the initial geometry was not set. Set the geometry and validate the geometry 
+     * Because the interface is shrinking, it is sufficient to validate only the first time the geometry
+     * is set.
+     */
+    if (m_geometry_set && m_validate_geom)
+        {
+        m_geom = std::make_shared<mpcd::detail::SphereGeometry>(start_R, V , m_bc ); 
+        validate();
+        m_geometry_set = false;
+        m_validate_geom = false;
+        }
+    /*
+     *Update the geometry radius
+     */
+    m_geom = std::make_shared<mpcd::detail::SphereGeometry>(end_R, V, m_bc );
+    
+    //stream according to base class rules
+    ConfinedStreamingMethod<mpcd::detail::SphereGeometry>::stream(timestep);
+
+    //calculating number of particles to remove from solvent density
+    const unsigned int N_remove = round((4.*M_PI/3.)*end_R*end_R*end_R*m_density);
+    const int N_evap = m_mpcd_pdata->getNGlobal() - N_remove;
+    std::cout << "Number of particles to evaporate is : " << N_evap << std::endl;
+
+    // get the compact array of indexes of bounced particles
+    unsigned int N_bounced = 0;
+    const unsigned int N = m_mpcd_pdata->getN();
+
+    ArrayHandle<unsigned char> h_bounced(m_bounced, access_location::host, access_mode::read);
+
+    bool overflowed = false;
+    do
+        {
+        if (overflowed)
+            {
+            GPUVector<unsigned char> bounced_index(N_bounced, m_exec_conf);
+            m_bounced_index.swap(bounced_index);
+            // resize m_bounced_index with swap idiom
+            }
+        ArrayHandle<unsigned char> h_bounced_index(m_bounced_index, access_location::host, access_mode::overwrite);
+        const unsigned int N_bounced_max = m_bounced_index.size();
+        N_bounced = 0;
+        for (unsigned int idx=0; idx < N; ++idx)
+            {
+            if (h_bounced.data[idx])
+                {
+                if (N_bounced < N_bounced_max)
+                    {
+                    h_bounced_index.data[N_bounced] = idx;
+                    }
+                ++N_bounced;
+                }
+            }
+        overflowed = N_bounced > N_bounced_max;
+        } while (overflowed);
+    std::cout << "N_bounced is : " << N_bounced << std::endl;
```

Comment:
Just FYI: if you were running this on multiple ranks, you would want to print only after the MPI scan below